packages/authenticator/amplify_authenticator/test/sign_up_form_test.dart

```diff
@@ -172,6 +219,60 @@ void main() {
           expect(passwordFieldErrorLine4, findsOneWidget);
         },
       );
+
+      testWidgets(
+        'trims the username field before validation',
+        (tester) async {
+          await tester.pumpWidget(
+            const MockAuthenticatorApp(
+              initialStep: AuthenticatorStep.signUp,
+            ),
+          );
+          await tester.pumpAndSettle();
+
+          final signInPage = SignUpPage(tester: tester);
+
+          await signInPage.enterUsername('user@example.com ');
+          await signInPage.enterPassword('Password123');
+
+          await signInPage.submitSignUp();
+
+          final usernameFieldError = find.descendant(
+            of: signInPage.usernameField,
+            matching: find.text('Invalid email format.'),
```

Comment:
question: This could cannot catch a validation error if we ever change the error message. Is there another assertion that can be made here? It's okay if not -- but worth looking into imo.